src/module.c

```diff
@@ -8264,16 +8301,24 @@ int moduleLateDefrag(robj *key, robj *value, unsigned long *cursor, long long en
 /* Attempt to defrag a module data type value. Depending on complexity,
  * the operation may happen immediately or be scheduled for later.
  *
- * Returns the number of defragmented blocks, or -1 if the operation
- * is to be scheduled for later.
+ * Returns 1 if the operation has been completed or 0 if it needs to
+ * be scheduled for late defrag.
  */
-
-long moduleDefragValue(robj *key, robj *value) {
+int moduleDefragValue(robj *key, robj *value, long *defragged) {
     moduleValue *mv = value->ptr;
     moduleType *mt = mv->type;
 
+    /* Try to defrag moduleValue itself regardless of whether or not
+     * defrag callbacks are provided.
+     */
+    moduleValue *newmv = activeDefragAlloc(mv);
+    if (newmv) {
+        (*defragged)++;
+        value->ptr = mv = newmv;
+    }
```

Comment:
all of that can be done in one line 8-)

Reply:
lots more can be done in one line. but why?